todo/src/terminal.rs

```diff
@@ -38,10 +47,60 @@ impl Terminal {
         self.write_stdout(&style(formatted_msg).green().to_string())
     }
 
-    fn user_intention(&mut self) -> Result<bool, TerminalError> {
-        self.write_stdout(&style("Do you want to input a new todo? (y/n)").blue().to_string())?;
-        let user_input = self.input();
-        Ok(matches!(user_input, Ok(input) if input == "y"))
+    pub fn alert_unrecognized(&mut self) -> Result<(), TerminalError> {
+        self.write_stdout(&style("Invalid option. Please type again").red().to_string())
+    }
+
+    pub fn clear_todo(&mut self, todo_list: &mut Todos) -> Result<(), TerminalError> {
+        todo_list.list.clear();
+        self.write_stdout(
+            &style("Successfully cleared all todos.")
+                .yellow()
+                .to_string(),
+        )?;
+        Ok(())
+    }
+
+    pub fn show_todo_list(&mut self, todo_list: &mut Todos) -> Result<(), TerminalError> {
+        if todo_list.list.is_empty() {
+            self.write_stdout(&style("Your current todo list is empty!").green().to_string())?;
+        } else {
+            self.write_stdout(&style("Your current todo list is:").green().to_string())?;
+            for todo in &todo_list.list {
+                self.show_todo(todo)?;
+            };
+        }
+        Ok(())
+    }
+
+    pub fn remove_todo(
+        &mut self,
+        todo_list: &mut Todos,
+        index: usize,
+    ) -> Result<(), TerminalError> {
+        todo_list.remove_todo(index)?;
+        self.write_stdout(&style("Successfully removed todo.").yellow().to_string())
+    }
+
+    pub fn user_intention(&mut self, todo_list: &mut Todos) -> Result<UserOptions, TerminalError> {
+        self.write_stdout(&style("Do you want to input a new todo? Type \"y\" to add a new todo or \"help\" to see all commands.").blue().to_string())?;
+        let user_input = self.input()?;
+
+        if let Some(index) = user_input.strip_prefix("rm ") {
+            let parsed_i = index
+                .parse::<usize>()
```

Comment:
```suggestion
                .parse()
```

Não necessariamente para mudar, mas esse turbofish não é necessário porque o compilador, nesse caso, irá conseguir inferir o tipo (porque você está usando ele dentro da variante `RemoveTodo`, que aceita apenas um `usize`).